staging/src/k8s.io/client-go/rest/client.go

```diff
@@ -100,11 +100,14 @@ type RESTClient struct {
 
 	// Set specific behavior of the client.  If not set http.DefaultClient will be used.
 	Client *http.Client
+
+	// RequestTimeout specifies a time limit for requests made by the HTTP Client
+	RequestTimeout time.Duration
 }
 
 // NewRESTClient creates a new RESTClient. This client performs generic REST functions
 // such as Get, Put, Post, and Delete on specified paths.
-func NewRESTClient(baseURL *url.URL, versionedAPIPath string, config ClientContentConfig, rateLimiter flowcontrol.RateLimiter, client *http.Client) (*RESTClient, error) {
+func NewRESTClient(baseURL *url.URL, versionedAPIPath string, config ClientContentConfig, rateLimiter flowcontrol.RateLimiter, client *http.Client, requestTimeout time.Duration) (*RESTClient, error) {
```

Comment:
worth to avoid a breaking change through `NewRESTClientWithTimeout` ?

Reply:
How many clients of this method do we have? 

I didn't have to fix many references, just two.

Reply:
Since client-go is a library, this change affects non k/k code as well.